### PR TITLE
feat: [PL-43365]: OIDC Support for CD Tasks

### PR DESCRIPTION
**Pull request by Raghav-Murali**
## Describe your changes

This PR adds support for OIDC in CD GCR Task. This PR contains the changes on the SAAS side pertaining to ng-manager and the pipeline service. 

As part of this PR the OIDC GCP library is also restructured and this is needed for CD task support as it involves Delegate support.

Tested this in local test environment and it works as expected.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56937)

### CODE REVIEW

File: 930-oidc-core/src/main/java/io/harness/oidc/gcp/dto/GcpOidcAccessTokenRequestDTO.java [hunk 1]
```diff
@@ -5,24 +5,23 @@
...
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonInclude.Include;
 import io.swagger.v3.oas.annotations.media.Schema;
 import javax.validation.constraints.NotEmpty;
 import javax.validation.constraints.NotNull;
-import lombok.Data;
-import lombok.NoArgsConstructor;
-import lombok.Setter;
+import lombok.*;
```
Comment (TarunUba1) [inline 930-oidc-core/src/main/java/io/harness/oidc/gcp/dto/GcpOidcAccessTokenRequestDTO.java:16]: nit: remove this * and lets import only required entities

File: 930-oidc-core/src/main/java/io/harness/oidc/gcp/dto/GcpOidcTokenRequestDTO.java [hunk 2]
```diff
@@ -5,24 +5,23 @@
...
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonInclude.Include;
 import io.swagger.v3.oas.annotations.media.Schema;
 import javax.validation.constraints.NotEmpty;
 import javax.validation.constraints.NotNull;
-import lombok.Data;
-import lombok.NoArgsConstructor;
-import lombok.Setter;
+import lombok.*;
```
Comment (TarunUba1) [inline 930-oidc-core/src/main/java/io/harness/oidc/gcp/dto/GcpOidcTokenRequestDTO.java:16]: nit: remove this * and lets import only required entities

File: 930-oidc-core/src/main/java/io/harness/oidc/gcp/dto/GcpOidcAccessTokenRequestDTO.java (hunk 1 above)
Comment (Raghav-Murali) [reply 930-oidc-core/src/main/java/io/harness/oidc/gcp/dto/GcpOidcAccessTokenRequestDTO.java]: Done.

File: 930-oidc-core/src/main/java/io/harness/oidc/gcp/dto/GcpOidcTokenRequestDTO.java (hunk 2 above)
Comment (Raghav-Murali) [reply 930-oidc-core/src/main/java/io/harness/oidc/gcp/dto/GcpOidcTokenRequestDTO.java]: Done

**Review by TarunUba1 — CHANGES_REQUESTED** [review]
Also please add the test files for the oidc module for better coverage and in description what all cases were tested as part of testing in general

**Comment by abhinavsingh1196** [comment]
cc: @puthrayaharness @Abhishek-CDC for review

**Comment by Raghav-Murali** [comment]
trigger uts

**Review by TarunUba1 — APPROVED** [review]

**Comment by Raghav-Murali** [comment]
trigger ut5, ut6, ut7, ut8, ut9

**Comment by Raghav-Murali** [comment]
trigger ut5, ut7

**Comment by Raghav-Murali** [comment]
trigger messagecheck

**Comment by Raghav-Murali** [comment]
trigger ut8, ut9

**Comment by Raghav-Murali** [comment]
trigger ut9

**Comment by Abhishek-CDC** [comment]
Looks good from GCR artifact side

**Review by Abhishek-CDC — APPROVED** [review]

**Review by VikasMaddukuriHarness — APPROVED** [review]

**Comment by TarunUba1** [comment]
trigger ut9

**Pull request merged**